rocAL/source/readers/image/cifar10_data_reader.cpp

```diff
@@ -141,9 +151,17 @@ int CIFAR10DataReader::release() {
 }
 
 void CIFAR10DataReader::reset() {
-    if (_shuffle)
-        std::random_shuffle(_file_names.begin() + _shard_start_idx_vector[_shard_id],
-                            _file_names.begin() + _shard_end_idx_vector[_shard_id]);
+    if (_shuffle) {
+        std::mt19937 rng1(_shard_id);
```

Comment:
std::mt19937 instance is costly. You should define it in the class instance and use the same for both)